### PR TITLE
Patch TaskExtensions bug in Azure.Storage

### PR DESCRIPTION
**Pull request by pakrym**
Contributes: https://github.com/Azure/azure-sdk-for-net/issues/14304

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs.Batch/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 12.3.1 (2020-08-18)
+
+### Fixed
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs.Batch/CHANGELOG.md:5]: We haven't used 3rd level headers anywhere else in our changelog.  Could we change this to regular text "Fix bug in TaskExtensions.EnsureCompleted method that causes it to unconditionally throw an exception in the environments with synchronization context"?

File: sdk/storage/Azure.Storage.Blobs.ChangeFeed/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 12.0.0-preview.4 (2020-08-18)
+
+### Fixed
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs.ChangeFeed/CHANGELOG.md:5]: We haven't used 3rd level headers anywhere else in our changelog.  Could we change this to regular text "Fix bug in TaskExtensions.EnsureCompleted method that causes it to unconditionally throw an exception in the environments with synchronization context"?

File: sdk/storage/Azure.Storage.Blobs/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 12.5.1 (2020-08-18)
+
+### Fixed
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/CHANGELOG.md:5]: We haven't used 3rd level headers anywhere else in our changelog.  Could we change this to regular text "Fix bug in TaskExtensions.EnsureCompleted method that causes it to unconditionally throw an exception in the environments with synchronization context"?

File: sdk/storage/Azure.Storage.Common/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 12.5.1 (2020-08-18)
+
+### Fixed
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Common/CHANGELOG.md:5]: We haven't used 3rd level headers anywhere else in our changelog.  Could we change this to regular text "Fix bug in TaskExtensions.EnsureCompleted method that causes it to unconditionally throw an exception in the environments with synchronization context"?

File: sdk/storage/Azure.Storage.Files.DataLake/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 12.3.1 (2020-08-18)
+
+### Fixed
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/CHANGELOG.md:5]: We haven't used 3rd level headers anywhere else in our changelog.  Could we change this to regular text "Fix bug in TaskExtensions.EnsureCompleted method that causes it to unconditionally throw an exception in the environments with synchronization context"?

File: sdk/storage/Azure.Storage.Files.Shares/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 12.3.1 (2020-08-18)
+
+### Fixed
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/CHANGELOG.md:5]: We haven't used 3rd level headers anywhere else in our changelog.  Could we change this to regular text "Fix bug in TaskExtensions.EnsureCompleted method that causes it to unconditionally throw an exception in the environments with synchronization context"?

File: sdk/storage/Azure.Storage.Queues/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 12.4.1 (2020-08-18)
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Queues/CHANGELOG.md:3]: We haven't used 3rd level headers anywhere else in our changelog.  Could we change this to regular text "Fix bug in TaskExtensions.EnsureCompleted method that causes it to unconditionally throw an exception in the environments with synchronization context"?

**Review by tg-msft — APPROVED** [review]
Looks good.

File: sdk/storage/Azure.Storage.Blobs.Batch/CHANGELOG.md (hunk 1 above)
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs.Batch/CHANGELOG.md:5]:
```suggestion
```

File: sdk/storage/Azure.Storage.Blobs.ChangeFeed/CHANGELOG.md (hunk 2 above)
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs.ChangeFeed/CHANGELOG.md:5]:
```suggestion
```

File: sdk/storage/Azure.Storage.Blobs/CHANGELOG.md (hunk 3 above)
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/CHANGELOG.md:5]:
```suggestion
```

File: sdk/storage/Azure.Storage.Common/CHANGELOG.md (hunk 4 above)
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/CHANGELOG.md:5]:
```suggestion
```

File: sdk/storage/Azure.Storage.Files.DataLake/CHANGELOG.md (hunk 5 above)
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/CHANGELOG.md:5]:
```suggestion
```

File: sdk/storage/Azure.Storage.Files.Shares/CHANGELOG.md (hunk 6 above)
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.Shares/CHANGELOG.md:5]:
```suggestion
```

File: sdk/storage/Azure.Storage.Queues/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,10 @@
...
+
+### Fixed
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Queues/CHANGELOG.md:5]:
```suggestion
```

**Review by tg-msft — COMMENTED** [review]
Adding suggestions to keep the Storage CHANGELOGs linear

**Review by seanmcc-msft — APPROVED** [review]

**Pull request merged**